### PR TITLE
[Merged by Bors] - feat: order properties of `Finsupp.single`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanCamCombi

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finsupp/Order.lean [hunk 1]
```diff
@@ -72,6 +84,26 @@ lemma coe_mono : Monotone (Finsupp.toFun : (ι →₀ α) → ι → α) := fun
...
+@[gcongr] protected alias ⟨_, GCongr.single_mono⟩ := single_le_single
+
+@[simp] lemma single_nonneg : 0 ≤ single i a ↔ 0 ≤ a := by classical exact Pi.single_nonneg
+@[simp] lemma single_nonpos : single i a ≤ 0 ↔ a ≤ 0 := by classical exact Pi.single_nonpos
+
+variable [OrderedAddCommMonoid β]
+
+lemma sum_le_sum_index [DecidableEq ι] {f₁ f₂ : ι →₀ α} {h : ι → α → β} (hf : f₁ ≤ f₂)
+    (hh : ∀ i ∈ f₁.support ∪ f₂.support, Monotone (h i))
+    (hh₀ : ∀ i ∈ f₁.support ∪ f₂.support, h i 0 = 0): f₁.sum h ≤ f₂.sum h := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Finsupp/Order.lean:101]:
```suggestion
    (hh₀ : ∀ i ∈ f₁.support ∪ f₂.support, h i 0 = 0) : f₁.sum h ≤ f₂.sum h := by
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

File: Mathlib/Data/Finsupp/Order.lean [hunk 2]
```diff
@@ -117,11 +149,24 @@ end Zero
 
 /-! ### Algebraic order structures -/
 
+section OrderedAddCommMonoid
+variable [OrderedAddCommMonoid α] {i : ι} {a b : α} {f : ι → κ} {g g₁ g₂ : ι →₀ α}
```
Comment (b-mehta) [inline Mathlib/Data/Finsupp/Order.lean:153]:
```suggestion
variable [OrderedAddCommMonoid α] {i : ι} {f : ι → κ} {g g₁ g₂ : ι →₀ α}
```
It doesn't look like these are used...

File: Mathlib/Data/Finsupp/Order.lean [hunk 3]
```diff
@@ -32,6 +34,16 @@ section Zero
...
+section OrderedAddCommMonoid
+variable [OrderedAddCommMonoid β] {f : ι →₀ α} {h₁ h₂ : ι → α → β}
+
+@[gcongr]
+lemma sum_le_sum (h : ∀ i ∈ f.support, h₁ i (f i) ≤ h₂ i (f i)) : f.sum h₁ ≤ f.sum h₂ :=
+  Finset.sum_le_sum h
+
+end OrderedAddCommMonoid
+
+
```
Comment (b-mehta) [inline Mathlib/Data/Finsupp/Order.lean:46]:
```suggestion
```

**Comment by b-mehta** [comment]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**